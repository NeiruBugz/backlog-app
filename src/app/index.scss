@import './styles/index.scss';

:root {
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu,
    Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 14px;
  line-height: 24px;
  font-weight: 400;

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
}

*,
*::after,
*::before {
  box-sizing: border-box;
  scroll-behavior: smooth;
}

#root {
  height: 100vh;
  width: 100%;
}

body {
  margin: 0;
  place-items: center;
  min-width: 320px;
  min-height: 100vh;
}

.card.image-full:before {
  background-color: transparent !important;
}
