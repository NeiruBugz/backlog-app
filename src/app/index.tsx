import { withProviders } from './providers';
import { Routing } from '@pages';

import './index.scss';

const App = () => <Routing />;

export default withProviders(App);
