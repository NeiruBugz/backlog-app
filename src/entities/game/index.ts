export * from './types';
export * from './ui/card';
export * from './ui/card-list';
