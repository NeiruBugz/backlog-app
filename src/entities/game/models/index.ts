import { createEffect, createEvent, createStore } from 'effector';
import { Filter, Game, UpdateGamePayload } from '../types';
import { HowLongToBeatEntry } from 'howlongtobeat';

const $games = createStore<Game[]>([]);
const addGame = createEvent<Game>('addGame');
const updateGameFx = createEffect<UpdateGamePayload, UpdateGamePayload, Error>();

const $filter = createStore<Filter>('all');
const setFilter = createEvent<Filter>('setFilter');

const $addPayload = createStore<HowLongToBeatEntry | null>(null);
const savePayload = createEvent<HowLongToBeatEntry>('savePayload');
const resetPayload = createEvent<null>('resetPayload');

$addPayload
  .on(savePayload, (state, payload) => ({ ...state, ...payload }))
  .on(resetPayload, (state) => {
    state = null;
    return state;
  });

$games
  .on(addGame, (state, payload) => [...state, payload])
  .on(updateGameFx, (state, { id, field }) => {
    const { key, value } = field;
    const gameIndex = state.findIndex((game) => game.id === id);
    if (gameIndex !== -1) {
      const game = state[gameIndex];
      game[key] = value;
      state[gameIndex] = game;
    }

    return [...state];
  });

$filter.on(setFilter, (state, payload) => (state = payload));

export { $games, $filter, setFilter, updateGameFx, $addPayload, resetPayload, savePayload };
