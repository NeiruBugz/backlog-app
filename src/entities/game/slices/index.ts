export * from './nano-filters';
export * from './nano-games';
export * from './nano-search';
