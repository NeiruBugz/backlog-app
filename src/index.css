:root {
  font-family: Inter, Avenir, Helvetica, Arial, sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  color: rgba(255, 255, 255, 0.87);
  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
  --ba-modal-index: 1000;
}

a, button {
  cursor: pointer;
}

.container {
  padding: 16px 20px 0;
}

.ba-button {
  border: 1px solid #242424;
  border-radius: 4px;
  background: transparent;
  text-align: center;
  padding: 4px 8px;
  transition: all .6ms ease-in-out;
}

.ba-button--primary {
  background-color: #242424;
  color: #fdfdfd;
}

.ba-button--primary:hover {
  background-color: transparent;
  color: #242424;
}

.ba-button--secondary:hover {
  border-color: transparent;
}

.ba-modal-overlay {
  width: 100vw;
  height: 100vh;
  position: absolute;
  z-index: var(--ba-modal-index, 1000);
  top: 0;
  left: 0;
  background-color: #24242442;
  display: flex;
  justify-content: center;
  align-items: center;
}

.ba-modal {
  background-color: #fdfdfd;
  display: flex;
  flex-direction: column;
  color: #242424;
}

.ba-modal-header {
  text-align: center;
  font-weight: bold;
  font-size: 20px;
  padding: 12px 24px 0;
}

.ba-modal-body {
  padding: 0 24px 12px;
}

.ba-label {
  display: block;
}

.ba-input {
  display: block;
}