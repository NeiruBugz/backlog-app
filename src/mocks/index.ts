import type { Game } from '@entities';

const cardProps: Game = {
  title: 'Persona 5 Royal',
  img: 'https://assets1.ignimgs.com/2020/02/14/persona-5-royal---button-fin-1581716582492.jpg',
  platform: 'Nintendo Switch',
  status: 'in-progress',
};

const gamesList: Game[] = [
  {
    title: 'God of War: Ragnar√∂k',
    img: '',
    platform: 'PlayStation 5',
    status: 'completed',
  },
  {
    title: 'The Last Of Us Part I',
    img: '',
    platform: 'PlayStation 5',
    status: 'completed',
  },
  {
    title: 'Forza Horizon 5',
    img: '',
    platform: 'Xbox',
    status: 'backlog',
  },
  cardProps,
];

const user = {
  username: 'NeiruBugz',
  avatarUrl: '',
};

export { cardProps, gamesList, user };
