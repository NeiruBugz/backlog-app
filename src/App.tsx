import 'normalize.css/normalize.css';
import './App.css';
import { useCallback, useState } from 'react';
import { Layout, Button, Space } from 'antd';

import { Card, AuthModal, Header } from '@components';
import { useEscapeKeyPress } from './hooks/useEscapeKeyPress';
import { useList } from 'effector-react';
import { addGame, gamesList } from '@store';
import { cardProps } from './mocks/data/card.mock';
import { Game } from './types/data/game';

const { Content, Sider } = Layout;

function App() {
  const [modalOpen, setModalOpen] = useState(false);
  useEscapeKeyPress(useCallback(() => setModalOpen(false), []));

  return (
    <>
      <Layout style={{ height: '100vh' }}>
        <Header setModalOpen={setModalOpen} />
        <Layout>
          <Sider width={200}>Sider</Sider>
          <Content>
            <div className="container">
              <Button type='primary' onClick={() => addGame(cardProps as Game)}>Add Persona 5</Button>
              <Space wrap>
                {useList(gamesList, (game) => <Card {...game} />)}
              </Space>
            </div>
          </Content>
        </Layout>
      </Layout>
      <AuthModal isOpen={modalOpen} onCancel={() => setModalOpen(false)} />
    </>
  );
}

export default App;
