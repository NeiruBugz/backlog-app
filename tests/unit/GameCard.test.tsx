import { render } from '@testing-library/react';
import React from 'react';
import { GameCard } from '../../src/entities/game';
import { cardProps } from '../../src/mocks';

describe('GameCard', () => {
  it('snapshots', () => {
    const { container } = render(<GameCard {...cardProps}/>);
    expect(container).toMatchSnapshot();
  });
});