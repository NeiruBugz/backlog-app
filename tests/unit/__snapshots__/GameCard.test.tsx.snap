// Vitest Snapshot v1

exports[`GameCard > snapshots 1`] = `
<div>
  <div
    class="card bg-base-100 shadow-xl image-full rounded-lg w-56 h-56 sm:w-64 sm:h-64 md:w-96 md:h-96"
  >
    <figure>
      <img
        class="w-96 object-fill rounded-lg"
        src="https://assets1.ignimgs.com/2020/02/14/persona-5-royal---button-fin-1581716582492.jpg"
      />
    </figure>
    <div
      class="card-body flex flex-col"
    >
      <h2
        class="card-title"
      >
        Persona 5 Royal
      </h2>
      <span
        class="py-2 px-4 h-8 max-h-10 flex justify-center items-center text-white text-md rounded bg-nintendo"
      >
        Nintendo Switch
      </span>
    </div>
  </div>
</div>
`;
